pkg/config/conversion/conversions.go

```diff
@@ -141,3 +151,39 @@ func NewCustomConverter(sourceVersion, targetVersion string, converter func(src,
 		customConverter: converter,
 	}
 }
+
+type singletonListConverter struct {
+	baseConversion
+	crdPaths []string
+	mode     Mode
+}
+
+// NewSingletonListConversion returns a new Conversion from the specified
+// sourceVersion of an API to the specified targetVersion and uses the
+// CRD field paths given in crdPaths to convert between the singleton
+// lists and embedded objects in the given conversion mode.
+func NewSingletonListConversion(sourceVersion, targetVersion string, crdPaths []string, mode Mode) Conversion {
+	return &singletonListConverter{
+		baseConversion: newBaseConversion(sourceVersion, targetVersion),
+		crdPaths:       crdPaths,
+		mode:           mode,
+	}
+}
+
+func (s *singletonListConverter) ConvertPaved(src, target *fieldpath.Paved) (bool, error) {
```

Comment:
Question: Does returned `bool` value have, or will have, any function? At the moment, it is equivalent to `error != nil`.

Reply:
Yes, the idea is an implementation of the `conversion.PavedConversion` may conditionally convert the source object. If it doesn't encounter any issues and if it leaves the target untouched, then we expect it to return `false, nil`.